browser/mapping.go

```diff
@@ -648,7 +648,22 @@ func mapBrowserContext(vu moduleVU, bc api.BrowserContext) mapping {
 		"setOffline":         bc.SetOffline,
 		"storageState":       bc.StorageState,
 		"unroute":            bc.Unroute,
-		"waitForEvent":       bc.WaitForEvent,
+		"waitForEvent": func(event string, optsOrPredicate goja.Value) *goja.Promise {
+			ctx := vu.Context()
+			return k6ext.Promise(ctx, func() (result any, reason error) {
+				resp, err := bc.WaitForEvent(event, optsOrPredicate)
+				if err != nil {
+					return nil, err //nolint:wrapcheck
+				}
+
+				p, ok := resp.(api.Page)
```

Comment:
I wonder if this verification should be moved to the `BrowserContext.waitForEvent` implementation, and in case it fails there, we would already receive an error here from the `bc.waitForEvent` call, that then can be propagated for the promise.

Did we move this here because we want to panic if the data received is not a page and we want to avoid panicking away from the mapping layer?

_Related with the discussion [here](https://github.com/grafana/xk6-browser/pull/1042/commits/2e2e0637c43c5ea5f7cd3838bdfec987e2e24866#r1332839382)._

Reply:
> I wonder if this verification should be moved to the BrowserContext.waitForEvent implementation